### PR TITLE
[bugfix] integration overrides now work with action destinations

### PR DESCRIPTION
**Pull request by chrisradek**
This PR updates how we resolve `integrations` passed to `analytics.load()` so that they now apply towards action destinations. Prior to this change they only applied to classic destinations.

The gist is now we merge the remote plugin settings from the CDN with the integration settings/overrides the user provides, whereas previously we were using the settings exclusively.

### CODE REVIEW

File: src/plugins/remote-loader/__tests__/index.test.ts [hunk 1]
```diff
@@ -339,11 +351,105 @@ describe('Remote Loader', () => {
...
+
+    const initOptions: InitOptions = {
+      integrations: {
+        remotePlugin: {
+          name: 'Chris Radek',
+        },
+      },
+    }
+
+    await AnalyticsBrowser.load(
```
Comment (chrisradek) [inline src/plugins/remote-loader/__tests__/index.test.ts:438]: I wanted to test this at the AnalyticsBrowser.load() level as well to make sure the integration options merging is working correctly.

File: src/plugins/remote-loader/index.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ export interface RemotePlugin {
   /** The UMD/global name the plugin uses. Plugins are expected to exist here with the `PluginFactory` method signature */
   libraryName: string
   /** The settings related to this plugin. */
-  settings: JSONValue
+  settings: JSONObject
```
Comment (chrisradek) [inline src/plugins/remote-loader/index.ts:17]: I checked how we build the settings that get served by the CDN and it's always an object.

File: src/plugins/segmentio/index.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ import standard from './fetch-dispatcher'
 import { normalize } from './normalize'
 import { scheduleFlush } from './schedule-flush'
 
-export interface SegmentioSettings {
+export type SegmentioSettings = {
```
Comment (silesky) [inline src/plugins/segmentio/index.ts:15]: Why the change?

Comment (chrisradek) [reply src/plugins/segmentio/index.ts:15]: TypeScript was complaining that `SegmentioSettings` wasn't a subset of the `JSONObject` shape (but it is since it's just an interface containing properties) - changing this from an interface to a type allowed TypeScript to recognize that.

**Review by pooyaj — APPROVED** [review]
:shipit:
